base/pkg/entry.jl

```diff
@@ -553,6 +553,20 @@ function resolve(
         end
         rethrow(err)
     end
+    imported = String[]
+    for (pkg,(ver1,ver2)) in changed
+        if ver1 !== nothing && ver2 !== nothing
```

Comment:
My understanding based on loop starting on line 528 is that for an install `ver1 === nothing` and for a removal `ver2 === nothing`.